cli/templateinit.go

```diff
@@ -23,29 +27,40 @@ func (*RootCmd) templateInit() *clibase.Cmd {
 			if err != nil {
 				return err
 			}
-			exampleNames := []string{}
-			exampleByName := map[string]codersdk.TemplateExample{}
+
+			optsToID := map[string]string{}
 			for _, example := range exampleList {
 				name := fmt.Sprintf(
 					"%s\n%s\n%s\n",
 					cliui.Styles.Bold.Render(example.Name),
 					cliui.Styles.Wrap.Copy().PaddingLeft(6).Render(example.Description),
 					cliui.Styles.Keyword.Copy().PaddingLeft(6).Render(example.URL),
 				)
-				exampleNames = append(exampleNames, name)
-				exampleByName[name] = example
+				optsToID[name] = example.ID
 			}
 
-			_, _ = fmt.Fprintln(inv.Stdout, cliui.Styles.Wrap.Render(
-				"A template defines infrastructure as code to be provisioned "+
-					"for individual developer workspaces. Select an example to be copied to the active directory:\n"))
-			option, err := cliui.Select(inv, cliui.SelectOptions{
-				Options: exampleNames,
-			})
-			if err != nil {
-				return err
+			// If the user didn't specify any template, prompt them to select one.
+			if templateIDArg == "" {
+				opts := keys(optsToID)
+				sort.Strings(opts)
+				_, _ = fmt.Fprintln(inv.Stdout, cliui.Styles.Wrap.Render(
+					"A template defines infrastructure as code to be provisioned "+
+						"for individual developer workspaces. Select an example to be copied to the active directory:\n"))
+				selected, err := cliui.Select(inv, cliui.SelectOptions{
+					Options: sort.StringSlice(keys(optsToID)),
+				})
+				if err != nil {
+					return err
+				}
+				templateIDArg = optsToID[selected]
+			}
+
+			selectedTemplate, ok := templateByID(templateIDArg, exampleList)
+			if !ok {
+				ids := values(optsToID)
+				sort.Strings(ids)
+				return xerrors.Errorf("Template ID %q does not exist!\nValid options are: %q", templateIDArg, ids)
```

Comment:
Not a biggie, but this was a bit of a brain twister for me since the name/id stored in the map seemed unintuitive. 😅 To me it would make sense to have `opts[id] = name`.

Reply:
Our goal is to have an example template ID that we can pass to `examples.Archive`. The choices passed to `cliui.Select` are just literally what gets displayed, so we have to map that back to ID. I'd honestly prefer if `cliui.Select` took a `map[string]string` so you could differentiate what gets displayed to what gets returned, but that's a bigger change than I want to make here.